docs/configuration.md

```diff
@@ -112,14 +114,15 @@ spec:
-| ------------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| `externalAddress`        | The loadbalancer address (for k0s controllers running behind a loadbalancer). Configures all cluster components to connect to this address and also configures this address for use  when joining new nodes to the cluster. |
-| `address`                | Local address on which to bind an API. Also serves as one of the addresses pushed on the k0s create service certificate on the API. Defaults to first non-local address found on the node.                                  |
-| `sans`                   | List of additional addresses to push to API servers serving the certificate.                                                                                                                                                |
-| `extraArgs`              | Map of key-values (strings) for any extra arguments to pass down to Kubernetes api-server process.                                                                                                                          |
-| `port`¹                  | Custom port for kube-api server to listen on (default: 6443)                                                                                                                                                                |
-| `k0sApiPort`¹            | Custom port for k0s-api server to listen on (default: 9443)                                                                                                                                                                 |
+| Element             | Description                                                                                                                                                                                                                  |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `address`           | Local address on which to bind an API. Also serves as one of the addresses pushed on the k0s create service certificate on the API. Defaults to first non-local address found on the node.                                   |
+| `onlyBindToAddress` | Whether to bind only to the IP address given by the `address` option, instead of binding to all addresses.                                                                                                                   |
```

Comment:
An alternative name could be `bindToAllInterfaces` which defaults to `true`. This would have to be implemented as a bool pointer though. `onlyBindToAddress` is a bit harder for me to understand.

Reply:
I read in other PRs that you generally prefer defaulting to false, so I'd assume other booleans are also like that. Not sure this is worth breaking the consistency, especially since this is more of a niche option.

Reply:
`bindExclusivelyToAddress`, `bindExclusive`, `bind(To)Address(Only)`, `addressBind`?

Somehow `onlyBindToAddress` sounds clunky to me.

Of course `address` (advertise) + `bindAddress: <address>` (bind) would be the best, I think there's an issue for that (#1880).

Reply:
> Somehow onlyBindToAddress sounds clunky to me.

I agree, I'm not super happy with the name, but at least it's descriptive.

> Of course address (advertise) + bindAddress: <address> (bind) would be the best

I actually don't agree here. I think `bindAddress` is not an option that should be offered, because the additional flexibility brings no benefit, only the risk that people misconfigure it.